test/test_linearizer.py

```diff
@@ -106,6 +108,19 @@ def test_sum_collapse(self):
     lin = Linearizer(sched[0].ast)
     assert not any(u.uop == UOps.LOOP for u in lin.linearize().uops), "found loop in sum collapse"
 
+  reduce_ops = (Tensor.max, Tensor.min, Tensor.sum)
+  @given(st.sampled_from(float_dtypes+int_dtypes), st.sampled_from(reduce_ops))
+  def test_reduce_acc(self, d:DType, op):
+    a = Tensor.rand(1024,1024, dtype=d)
+    out = op(a)
+
+    ast = [si for si in out.lazydata.schedule() if si.ast.op not in LoadOps][0].ast
+    reduceop = [op for op in ast.get_lazyops() if op.op in ReduceOps][0]
+    uops = Linearizer(ast).linearize().uops
+    phi = [u for u in uops if u.uop == UOps.PHI][0]
+
+    assert phi.dtype == phi.vin[0].dtype == phi.vin[1].dtype == get_lazyop_info(reduceop).dtype
```

Comment:
this does not test the case that reduceop has a different type from vin[0]

Reply:
i removed it i dont think it's actually correct
https://github.com/tinygrad/tinygrad/pull/2625/commits/86e74d796023e71122a09b9355bef493da6ae086

Reply:
we probably need some changes at the ops level to make a float/half/float kernel work

Reply:
oh okay i guess that's fine for now

Reply:
added back but skipped with a TODO